src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -13109,5 +13161,512 @@ public static void Main()
+        public void LambdaDefaultParameter_InterpolatedStringHandler()
+        {
+            var source = """
+using System.Runtime.CompilerServices;
+using System;
+using System.Text;
+
+[InterpolatedStringHandler]
+public ref struct CustomInterpolatedStringHandler
+{
+    StringBuilder builder;
+
+    public CustomInterpolatedStringHandler(int literalLength, int formattedCount) { builder = new StringBuilder(literalLength); }
+    public void AppendLiteral(string s) { builder.Append(s + "<<CustomInterpolatedStringHandler>>"); }
+    public void AppendFormatted<T>(T t) { builder.Append(t?.ToString()); }
+    internal string GetFormattedText() => builder.ToString();
+}
+
+public class Program
+{
+     public static void Main()
+     {
+        int i = 0;
+        var lam = (CustomInterpolatedStringHandler h = $"i: {i}") =>
+        {
+            Console.WriteLine(h.GetFormattedText());
+        };
+        lam();
+     }
+}
+""";
+            CreateCompilation(source, targetFramework: TargetFramework.Net60).VerifyDiagnostics(
+                // (21,56): error CS1736: Default parameter value for 'h' must be a compile-time constant
+                //         var lam = (CustomInterpolatedStringHandler h = $"i: {i}") =>
+                Diagnostic(ErrorCode.ERR_DefaultValueMustBeConstant, @"$""i: {i}""").WithArguments("h").WithLocation(21, 56));
+        }
+
+        [Fact]
+        public void LambdaWithDefault_InvalidConstantConversion()
+        {
+            TestDiagnosticsInMain(
+@"var lam = (string s = 1) => { };", expectedDiagnostics:
+                // (6,27): error CS1750: A value of type 'int' cannot be used as a default parameter because there are no standard conversions to type 'string'
+                //         var lam = (string s = 1) => { };
+                Diagnostic(ErrorCode.ERR_NoConversionForDefaultParam, "s").WithArguments("int", "string").WithLocation(6, 27));
+        }
+
+        [Fact]
+        public void LambdaWithDefault_NonConstantNonLiteral()
+        {
+            var source = """
+class Program
+{
+    // Named delegate has required parameter x
+    public static int f(int x) => 2 * x;
+    public static void Main()
+    {
+        // lambda has optional parameter x
+        var lam = (int x = f(1000)) => { };
+    }
+}
+""";
+            CreateCompilation(source).VerifyDiagnostics(
+                // (8,28): error CS1736: Default parameter value for 'x' must be a compile-time constant
+                //         var lam = (int x = f(1000)) => { };
+                Diagnostic(ErrorCode.ERR_DefaultValueMustBeConstant, "f(1000)").WithArguments("x").WithLocation(8, 28));
+        }
+
+        [ConditionalFact(typeof(NoIOperationValidation))]
+        public void LambdaWithDefault_NonConstantLiteral_InterpolatedString()
+        {
+            var source = """
+class Program
+{
+    // Named delegate has required parameter x
+    public static int f(int x) => 2 * x;
```

Comment:
Is this used in the test?

Reply:
Good catch, it isn't. I'm not sure why it's there (this was Adam's code), so I've removed it: 75e794341641ec666c637e9b9a03a039e2d3f2c2